src/hotspot/cpu/arm/stubGenerator_arm.cpp

```diff
@@ -635,13 +635,15 @@ class StubGenerator: public StubCodeGenerator {
     Register result_hi = R1;
     Register src       = R0;
 
-    if (!os::is_MP()) {
-      __ ldmia(src, RegisterSet(result_lo, result_hi));
-      __ bx(LR);
-    } else if (VM_Version::supports_ldrexd()) {
+    if (VM_Version::supports_ldrexd()) {
       __ ldrexd(result_lo, Address(src));
       __ clrex(); // FIXME: safe to remove?
       __ bx(LR);
```

Comment:
bx(LR) is common for all 3 branches, so it might be better to move it out

Reply:
It seems that `bx(LR)` near `stop` is redundant. But I don't see why not move it outside the block. See new commit.

Reply:
@dsamersoff, are you happy with new version?

Reply:
Yes